bliss.js

```diff
@@ -1,6 +1,38 @@
 (function() {
 "use strict";
 
+function overload(callback, start) {
+	start = start === undefined ? 1 : start;
+
+	return function() {
+		var isSingleParam = $.type(arguments[start]) === 'object', 
+			argsArray = Array.prototype.slice.call(arguments),
+			end = start + 1,
+			obj = {}, 
+			ret,
+			paramsBefore = [],
+			paramsAfter = [];
+
+		if (isSingleParam) {
+			obj = arguments[start];
+		}
+		else {
+			end += 1;
+			obj[arguments[start]] = arguments[start + 1];
+		}
+		
+		paramsBefore = argsArray.slice(0, start);
+		paramsAfter = argsArray.slice(end);
+
+		for (var key in obj) {
+			ret = callback.apply(
+				this, paramsBefore.concat([key, obj[key]], paramsAfter)
```

Comment:
well I need the before & after, as some of these methods have params before the overloaded ones, and some after. 

I could do it with slice, but IMO its easier to read to store the before and after in vars instead of extracting the params and concatenating all at once. I could change that though if you feel otherwise.